imap_data_access/file_validation.py

```diff
@@ -397,3 +397,227 @@ def construct_path(self) -> Path:
+         that if end_date is not provided, the file is valid until a file with a
+         later start_date and no end_date.
+        <version>: This stores the data version for this product, format: vXXX
+
+        Parameters
+        ----------
+        filename : str | Path
+            Ancillary data filename or file path.
+        """
+        self.filename = Path(filename)
+        self.data_dir = imap_data_access.config["DATA_DIR"]
+
+        try:
+            split_filename = self.extract_filename_components(self.filename)
+        except ValueError as err:
+            raise self.InvalidAncillaryFileError(
+                f"Invalid filename. Expected file to match format: "
+                f"{imap_data_access.ANCILLARY_FILENAME_CONVENTION}"
+            ) from err
+
+        self.mission = split_filename["mission"]
+        self.instrument = split_filename["instrument"]
+        self.description = split_filename["description"]
+        self.start_date = split_filename["start_date"]
+        self.end_date = split_filename["end_date"]
+        self.version = split_filename["version"]
+        self.extension = split_filename["extension"]
+
+        self.error_message = self.validate_filename()
+        if self.error_message:
+            raise self.InvalidAncillaryFileError(f"{self.error_message}")
+
+    @classmethod
+    def generate_from_inputs(
+        cls,
+        instrument: str,
+        description: str,
+        version: str,
+        extension: str,
+        start_time: str,
+        end_time: str | None = None,
+    ) -> AncillaryFilePath:
+        """Generate filename from given inputs and return a AncillaryFilePath instance.
+
+        This can be used instead of the __init__ method to make a new instance:
+        ```
+        ancillary_file_path = AncillaryFilePath.generate_from_inputs("mag",
+        "mag-rotation-matrices", "20240213", "v001")
+        full_path = ancillary_file_path.construct_path()
+        ```
+
+        Parameters
+        ----------
+        description : str
+            The descriptor for the ancillary filename.
+        instrument : str
+            The instrument for the filename.
+        start_time: str, optional
+            The start time for the filename. If not provided
+            it is assumed to be valid for all times.
+        end_time: str, optional
+            The end time for the filename. If not provided,
+            the file is valid until a file with a later
+            start_date and no end_date.
+        version : str
+            The version of the data.
+        extension : str
+            The extension type of the file.
+
+        Returns
+        -------
+        str
+            The generated filename
+        """
+        time_field = start_time
+        if end_time:
+            time_field += end_time  # I think...
+        filename = (
+            f"imap_{instrument}_{description}_{time_field}_" f"{version}.{extension}"
+        )
```

Comment:
Isn't the `end_time` just a `YYYYMMDD` formatted date (or `None`) provided by the user? So I think you could just do:

```suggestion
        if end_time:
            filename = (
                f"imap_{instrument}_{description}_{start_time}-{end_time}_{version}.{extension}"
            )
        else:
           filename = (
                f"imap_{instrument}_{description}_{start_time}_{version}.{extension}"
            )
```